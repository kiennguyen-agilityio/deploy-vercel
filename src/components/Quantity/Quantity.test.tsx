import { fireEvent, render } from '@testing-library/react';
import QuantityControl from './Quantity';

const defaultProps = {
  quantity: 10,
  onQuantityChange: jest.fn(),
};

const setup = (props = {}) => {
  const mergedProps = { ...defaultProps, ...props };
  return render(<QuantityControl {...mergedProps} />);
};

describe('QuantityControl component', () => {
  it('renders correctly', () => {
    const { container } = setup();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with quantity of value =  0', () => {
    const { container } = setup({ quantity: 0 });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('calls onQuantityChange with -1 when "-" button is clicked', () => {
    const onQuantityChange = jest.fn();
    const { getByText } = setup({ onQuantityChange });

    const minusButton = getByText('-');
    fireEvent.click(minusButton);

    expect(onQuantityChange).toHaveBeenCalledWith(-1);
  });

  it('calls onQuantityChange with 1 when "+" button is clicked', () => {
    const onQuantityChange = jest.fn();
    const { getByText } = setup({ onQuantityChange });

    const plusButton = getByText('+');
    fireEvent.click(plusButton);

    expect(onQuantityChange).toHaveBeenCalledWith(1);
  });
});
