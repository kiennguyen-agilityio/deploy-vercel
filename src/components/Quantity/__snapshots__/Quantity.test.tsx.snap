// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityControl component renders correctly 1`] = `
<div
  class="quantity-group"
>
  <p>
    Quantity
  </p>
  <div
    class="quantity"
  >
    <button
      class="btn btn-tertiary btn-color-success btn-md  "
      type="button"
    >
      <span>
        -
      </span>
    </button>
    <div
      class="quantity-line-number"
    >
      <span>
        10
      </span>
      <div
        class="divider"
      />
    </div>
    <button
      class="btn btn-tertiary btn-color-warning btn-md  "
      type="button"
    >
      <span>
        +
      </span>
    </button>
  </div>
</div>
`;

exports[`QuantityControl component renders correctly with quantity of value =  0 1`] = `
<div
  class="quantity-group"
>
  <p>
    Quantity
  </p>
  <div
    class="quantity"
  >
    <button
      class="btn btn-tertiary btn-color-success btn-md btn-disabled "
      disabled=""
      type="button"
    >
      <span>
        -
      </span>
    </button>
    <div
      class="quantity-line-number"
    >
      <span>
        0
      </span>
      <div
        class="divider"
      />
    </div>
    <button
      class="btn btn-tertiary btn-color-warning btn-md  "
      type="button"
    >
      <span>
        +
      </span>
    </button>
  </div>
</div>
`;
